.github/workflows/release-backwards-compatiblity.yml

```diff
@@ -0,0 +1,83 @@
+name: Backwards Compatibility Test
+
+on:
+  pull_request:
+    branches:
+      - master
+
+env:
+  SELENIUM_GRID_PROVIDER: saucelabs
+  CLOUD_WATCH_METRIC: false
+  TEST_TYPE: Github-Action
+  SAUCE_USERNAME: ${{secrets.SAUCE_USERNAME}}
+  SAUCE_ACCESS_KEY: ${{secrets.SAUCE_ACCESS_KEY}}
+  MESSAGING_USER_ARN: ${{secrets.MESSAGING_USER_ARN}}
+
+jobs:
+  release-integ-test:
+    name: Run Backwards Compatible Integration Tests with Release Tarball
+    runs-on: ubuntu-latest
+    if: ${{ github.head_ref == 'release' }}
```

Comment:
For learning: this `github.head_ref` is basically the current branch right? We do not check this out here at this line or earlier. So, do we manually run these workflows on release branches? As far as I understand, we are generating the current versions tarball from master branch itself and checking out the previous version and then trying to run the integration tests on that checked out versions demo so could you please help me understand how the release branch comes into play here?

Reply:
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context

The head_ref points to the name of the branch which is submitting the PR. The base_ref is the target branch (where it will be merged into after the PR is merged.) In this case, this workflow will only run if the PR is coming from the release branch.

This workflow should run the integ tests for the release branch - the main difference between this workflow and our regular integration workflows is that this workflow uses a `generated tarball` in the meeting demo, whereas the regular integration tests just have "file: ../.." as the dependency on the chime sdk.

Either we can switch all of our integration tests to use tarball like this workflow, or we should have a separate workflow (regular integ tests use "file: ../.." and another workflow uses tarball).

Reply:
Cool, thanks for the explanation, I had searched on github head refs and found that it represent the current branch so got confused here. Thanks for clarifying.